src/main/java/it/reply/orchestrator/service/deployment/providers/MarathonServiceImpl.java

```diff
@@ -670,7 +670,35 @@ public void finalizeDeploy(DeploymentMessage deploymentMessage) {
 
   @Override
   public void cleanFailedDeploy(DeploymentMessage deploymentMessage) {
-    // DO NOTHING
+    doUndeploy(deploymentMessage);
+  }
```

Comment:
This do not respect the `isLastProvider` and `isKeepLastAttempt` logic

Reply:
the cycle of tests on the various providers is however carried out. keeplastattempt only makes sense for IM deployments

Reply:
>keeplastattempt only makes sense for IM deployments

why does it not?

Reply:
keeplastattempt serves when the deployment of large infrastructures is required in order to preserve a possible partial deployment (eg 150 VM out of 200), in other cases it is not needed

Reply:
moreover, the "keeplastattempt" logic is absolutely not implemented in those providers